src/ble/BleLayer.cpp

```diff
@@ -104,15 +107,24 @@ class BleEndPointPool
     {
         typedef union
         {
-            uint8_t Data[sizeof(BLEEndPoint)];
-            uint64_t ForceAlignment;
+            uint8_t Data[((sizeof(BLEEndPoint) + BLE_ENDPOINT_ALIGNMENT - 1) / BLE_ENDPOINT_ALIGNMENT) * BLE_ENDPOINT_ALIGNMENT];
+            BLE_ENDPOINT_ALIGNMENT_TYPE ForceAlignment;
         } PoolElement;
 
-        static uint8_t sEndPointPool[sizeof(PoolElement) * BLE_LAYER_NUM_BLE_ENDPOINTS];
+        static union
+        {
+            uint8_t Pool[sizeof(PoolElement) * BLE_LAYER_NUM_BLE_ENDPOINTS];
+            BLE_ENDPOINT_ALIGNMENT_TYPE ForceAlignment;
+        } sEndPointPool;
+
+        // Assert sEndPointPool is aligned
+        nlASSERT(((uintptr_t) &sEndPointPool % BLE_ENDPOINT_ALIGNMENT) == 0);
+        // Assert sEndPointPool elements are aligned
+        nlASSERT(((uintptr_t) (sEndPointPool.Pool + sizeof(PoolElement)) % BLE_ENDPOINT_ALIGNMENT) == 0);
```

Comment:
I would prefer that we not incur a runtime overhead for checking the alignment on every Get() call.  It isn't worth it.

Reply:
Good point, I originally used static asserts, but there's no point of runtime asserts here.